### PR TITLE
Allow linux users to close windows without quitting rancher.

### PR DESCRIPTION
**Pull request by marknefedov**
Added option to enable "minimize on close" on Linux, instead of always closing Rancher desktop instance.

### CODE REVIEW

**Comment by adamkpickering** [comment]
Thanks for the PR 😃 it looks like you've taken the time to polish it, which is really appreciated. You've touched on a nuanced issue that we've had on the back burner for some time.

There are three ways RD is packaged for Linux: as an .rpm, as a .deb, and as an AppImage. When RD is packaged as an AppImage, it does not integrate with the system tray. This means that if one were to close the RD preferences window, there would be no way to open that instance of RD again. I have been wanting to investigate this further myself, but haven't had time. But this is the original reason that we made RD exit when all windows are closed on Linux. For more info see:
- #824
- #2129 (which is kind of a duplicate but covers Rancher Dashboard access as well)

Note that the system tray is also an issue on other platforms:
- #243

As you no doubt had in mind, having the `minimizeOnClose` flag means that the user can opt into not exiting when the window is closed. However, we want to be careful about adding things to settings.json.

Long story short, I need to discuss this further with the team before considering merging this. It looks like we have to do some design work around minimizing, the system tray et cetera.

**Comment by adamkpickering** [comment]
Setting this to a draft PR so that we don't talk about it at every standup.

**Comment by adamkpickering** [comment]
A quick update:

- The team discussed how Rancher Desktop should behave with regard to closing the main window. I wrote up the results in #3144 if you are curious to read them. But FWIW it's basically what you did in this PR, plus some extra stuff for when RD is run as an AppImage.

- The implementation of this behavior is blocked by an upstream bug in electron. For more info please see #2437. The electron bug is [here](https://github.com/electron/electron/issues/35680).

Until the electron bug is fixed, we're blocked I'm afraid :disappointed:

**Comment by adamkpickering** [comment]
I'm going to close this PR since we have incorporated the related features into main. @marknefedov I apologize for not merging this specific PR - we ended up having to slice the problem up in a different way. Your contribution wasn't wasted: it got the ball rolling, so thank you for that!
